udao/optimization/soo/mogd.py

```diff
@@ -751,19 +771,39 @@ def objective_loss(
             raise NotImplementedError("Objective with only one bound is not supported")
         return loss
 
-    def compute_loss(self, problem: co.SOProblem, input_data: Any) -> th.Tensor:
-        obj_output = problem.objective.function(input_data)
+    def _compute_loss(
+        self, problem: co.SOProblem, input_data: Union[UdaoInput, Dict]
+    ) -> Tuple[th.Tensor, float, float, int, bool]:
+        obj_output = (
+            problem.objective.function(input_data)  # type: ignore
+            if isinstance(input_data, UdaoInput)
+            else problem.objective.function(**input_data)
```

Comment:
Sorry I missed this. 
I like the fact that the lower level function makes the distinction between the two kinds of inputs. 

If we have time (other PR) it would make the logic clearer to have a single optimization loop that calls different specialized functions that can make a difference between different kinds of inputs.